### PR TITLE
Feat/create react app v4

### PR DESCRIPTION
**Pull request by kcwinner**
*Issue #, if available:* [158](https://github.com/eladb/projen/issues/158)

*Description of changes:*
* Adds new project `react`
* Adds new project `react-ts`
* Adds a new option to tsconfig options that react expects - `allowSyntheticDefaultImports`

Something to discuss - should the output files use:
1. `fs.writeFileSync(path.join(srcdir, 'logo.svg'), logoSvg.join('\n'));` - just joins the string array
1. `new ReactAppTest(this.project, path.join(srcdir, 'App.test.' + this.fileExt));` - extends `FileBase`

I left both options as examples in the code. Either to swap either way. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by eladb — APPROVED** [review]

**Review by eladb — CHANGES_REQUESTED** [review]
Sorry haven't seen the comment yet

**Review by eladb — COMMENTED** [review]
Apologies I merged to quickly. Let's discuss your question over a subsequent PR

File: src/react.ts [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+    // js or ts not jsx/tsx
+    fs.writeFileSync(path.join(srcdir, 'reportWebVitals.' + this.fileExt.replace('x', '')), reportWebVitalsJs.join('\n'));
+    new ReactSetupTests(this.project, path.join(srcdir, 'setupTests.' + this.fileExt.replace('x', '')));
+
+    const assetsdir = path.join(outdir, 'public');
+    if (fs.pathExistsSync(assetsdir) && fs.readdirSync(assetsdir)) return;
+
+    fs.mkdirpSync(assetsdir);
+
+    const indexHtml = [
```
Comment (eladb) [inline src/react.ts:334]: Might make sense to store this in a separate file and fs.readFile

File: src/react.ts [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+      '      getLCP(onPerfEntry);',
+      '      getTTFB(onPerfEntry);',
+      '    });',
+      '  }',
+      '}',
+      '',
+      'export default reportWebVitals;',
+    ];
+
+    fs.mkdirpSync(srcdir);
```
Comment (eladb) [inline src/react.ts:316]: See the PR @mbonig is working on. I think we'll have something like `SampleDir` which will accept multiple files.

**Review by eladb — COMMENTED** [review]
See response

File: src/react.ts (hunk 2 above)
Comment (kcwinner) [reply src/react.ts:316]: I can go back and clean this up once that PR is in!

Comment (eladb) [reply src/react.ts:316]: Thanks

**Pull request merged**